scio-core/src/main/scala/com/spotify/scio/VersionUtil.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2016 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio
+
+import com.google.api.client.http.javanet.NetHttpTransport
+import com.google.api.client.http.{GenericUrl, HttpRequest, HttpRequestInitializer}
+import com.google.api.client.json.JsonObjectParser
+import com.google.api.client.json.jackson2.JacksonFactory
+import org.slf4j.LoggerFactory
+
+import scala.collection.mutable
+import scala.util.Try
+
+private object VersionUtil {
+
+  case class SemVer(major: Int, minor: Int, rev: Int, snapshot: Int) {
+    def toInt: Int = major * 1000000 + minor * 10000 + rev * 100 + snapshot
+  }
+
+  private val TIMEOUT = 3000
+  private val url = "https://api.github.com/repos/spotify/scio/releases"
+  private val pattern = """v?(\d+)\.(\d+).(\d+)(-SNAPSHOT)?""".r
+  private val logger = LoggerFactory.getLogger("VersionUtil")
+
+  private def getLatest: Option[String] = Try {
+    val transport = new NetHttpTransport()
+    val response = transport
+      .createRequestFactory(new HttpRequestInitializer {
+        override def initialize(request: HttpRequest) = {
+          request.setConnectTimeout(TIMEOUT)
+          request.setReadTimeout(TIMEOUT)
+          request.setParser(new JsonObjectParser(new JacksonFactory))
+        }
+      })
+      .buildGetRequest(new GenericUrl(url))
+      .execute()
+      .parseAs(classOf[java.util.List[Object]])
+    val latest = response.iterator().next().asInstanceOf[java.util.Map[String, AnyRef]]
+    latest.get("tag_name").toString
+  }.toOption
+
+  private def parseVersion(version: String): SemVer = {
+    val m = pattern.findFirstMatchIn(version).get
+    // higher value for no "-SNAPSHOT"
+    val snapshot = if (m.group(4) == "-SNAPSHOT") 0 else 1
+    SemVer(m.group(1).toInt, m.group(2).toInt, m.group(3).toInt, snapshot)
+  }
+
+  def checkVersion(current: String, latest: Option[String]): Seq[String] = {
+    val b = mutable.Buffer.empty[String]
```

Comment:
Yes but that'd require 2 `if () Seq("...") else Nil` blocks which IMO is slightly more cryptic.